src/tracer_client.rs

```diff
@@ -519,4 +520,51 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
```

Comment:
we should cover edge cases but for now is okay

Reply:
What edge case do you mean?